src/sciline/visualize.py

```diff
@@ -7,22 +7,27 @@
 from .pipeline import Graph
 
 
-def to_graphviz(graph: Graph) -> Digraph:
+def to_graphviz(graph: Graph, **kwargs: Any) -> Digraph:
     """
     Convert output of :py:class:`sciline.Pipeline.get_graph` to a graphviz graph.
 
     Parameters
     ----------
     graph:
         Output of :py:class:`sciline.Pipeline.get_graph`.
+    kwargs:
+        Keyword arguments passed to :py:class:`graphviz.Digraph`.
     """
-    dot = Digraph(strict=True)
+    dot = Digraph(strict=True, **kwargs)
     for p, (p_name, args, ret) in _format_graph(graph).items():
+        dot.node(ret, ret, shape='rectangle')
+        # Do not draw dummy providers created by Pipeline when setting instances
+        if p_name == 'Pipeline.__setitem__.<locals>.<lambda>':
+            continue
```

Comment:
I don't think so, since their existence is just an implementation detail: We could implement this without making lambdas or other callables. So I think just drawing the rectangles for the types is correct?

Reply:
Yeah right. It will not be able to draw the graph if there is not enough providers so I think it makes sense now. Thanks...!